.github/workflows/runtime-reproducibility.yml

```diff
@@ -0,0 +1,73 @@
+name: verify runtime stack reproducibility
+
+on:
+    workflow_dispatch:
+    pull_request:
+      paths:
+        - 'node-installer/**'
+        - 'packages/**'
+
+jobs:
+  os-matrix:
+    strategy:
+      matrix:
+        os: [ubuntu-22.04, ubuntu-20.04]
+        build-target: [contrast-node-installer-image]
+    # Usually we would define the matrix outputs here, but as GitHub Actions don't seem to allow per-combination outputs,
+    # we'll write the outputs without defining them here. See https://github.com/orgs/community/discussions/17245#discussioncomment-3814009.
+    runs-on: ${{ matrix.os }}
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - uses: ./.github/actions/setup_nix
+        with:
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
+          cachixToken: ${{ secrets.CACHIX_AUTH_TOKEN }}
+      - uses: nicknovitski/nix-develop@a2060d116a50b36dfab02280af558e73ab52427d # v1.1.0
+      - name: Build
+        id: build
+        run: |
+            nix build .#${{ matrix.build-target }} --option substituters https://cache.nixos.org --builders ""
+            reference_checksum="$(cat result/index.json | jq  -r '.manifests[0].digest')"
+            echo "reference-checksum-${{ matrix.os }}-${{ matrix.build-target}}=$reference_checksum" >> "$GITHUB_OUTPUT"
+
+            nix build .#${{ matrix.build-target }} --rebuild --option substituters https://cache.nixos.org --builders "" -o rebuild
```

Comment:
Using nix build --rebuild, we will only rebuild the referenced derivation, not its dependencies. Why are we doing this? I mean sure, makes sense to check that building the OCI is reproducible, but I think we'd expect most issues further down. Not saying this is bad, but we should document why we are doing this and what guarantees it provides us.

Reply:
Didn't think about this

Reply:
>  but we should document why we are doing this and what guarantees it provides us.

is still unaddressed.